meeteval/wer/wer/cp.py

```diff
@@ -157,27 +171,27 @@ def split(words):
             else:
                 raise TypeError(type(words), words)
 
-        if isinstance(x, dict):
-            return {k: split(v) for k, v in x.items()}
-        elif isinstance(x, list):
-            return [split(e) for e in x]
-        elif isinstance(x, tuple):
-            return [split(e) for e in x]
-        elif isinstance(x, meeteval.io.stm.STM):
-            unique_filenames = {line.filename for line in x}
-            assert len(unique_filenames) == 1, (len(unique_filenames), unique_filenames, x)
+        if isinstance(x, meeteval.io.stm.STM):
+            assert len(x.filenames()) == 1, (len(x.filenames()), x.filenames(), x)
             return transcription_to_words(x.grouped_by_speaker_id())
         else:
-            raise TypeError(x)
+            return _map(split, x)
 
-    return _cp_word_error_rate(
+    return cp_error_rate(
         transcription_to_words(reference),
         transcription_to_words(hypothesis),
-        distance_fn=levenshtein_distance,
-        siso_error_rate=_siso_error_rate,
     )
 
 
+def cp_word_error_rate_stm(reference_stm: 'STM', hypothesis_stm: 'STM') -> 'Dict[str, CPErrorRate]':
+    """
+    TODO: doc
+    TODO: return some kind of summary type?
```

Comment:
I think a dict should be fine.
To get the summary, we may want to support `sum(cp_word_error_rate_stm().values())` (I am not sure, if I pushed that code already, probably not).

Reply:
I think `sum` already works on `ErrorRate`

Reply:
Ok, then I would return a dict. When the user wants to have the summary, it is easy to use sum.
We could give a hint for this in the doctext.